pkg/appgw/internaltypes.go

```diff
@@ -127,18 +134,15 @@ func generateFrontendPortName(port Port) string {
 }
 
 func generateListenerName(listenerID listenerIdentifier) string {
-	if listenerID.UsePrivateIP {
-		return formatPropName(fmt.Sprintf("%s%s-%v%v-privateip", agPrefix, prefixListener, formatHostname(listenerID.HostName), listenerID.FrontendPort))
-	}
-	return formatPropName(fmt.Sprintf("%s%s-%v%v", agPrefix, prefixListener, formatHostname(listenerID.HostName), listenerID.FrontendPort))
+	return formatPropName(fmt.Sprintf("%s%s-%s", agPrefix, prefixListener, utils.GetHashCode(listenerID)))
```

Comment:
What do you think of the following ?
```go
hostName := listenerID.HostNames[0]
if len(listenerID.HostNames) > 1 {
    hostName := utils.GetHashCode(listenerID)
}

if listenerID.UsePrivateIP {
    return formatPropName(fmt.Sprintf("%s%s-%v%v-privateip", agPrefix, prefixListener, formatHostname(hostName), listenerID.FrontendPort))	
}	

return formatPropName(fmt.Sprintf("%s%s-%v%v", agPrefix, prefixListener, formatHostname(hostName), listenerID.FrontendPort))
```

Reply:
Resolving this as we need to move away from having hostname in "name" of listener as it can have special characters and application gateway doesnot support it